### PR TITLE
Add support for Terraform 1.3

### LINKED ISSUE #2309: Upgrade Terragrunt to support Terraform 1.3
Upgrade Terragrunt to support Terraform 1.3

References:

https://github.com/hashicorp/terraform/blob/v1.3/CHANGELOG.md

https://www.hashicorp.com/blog/terraform-1-3-improves-extensibility-and-maintainability-of-terraform-modules

### PR DESCRIPTION
**Pull request by denis256**
## Description

Included changes:
  * upgraded circle ci image to run with Terraform 1.3
  * updated Windows tests to download and install Terraform 1.3
  * updated documentation to reflect usage of Terraform 1.3

Fixes #2309.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Updated supported Terraform version to 1.3

### CODE REVIEW

**Review by autero1 — DISMISSED** [review]
This LGTM!

File: config/config_helpers.go [hunk 1]
```diff
@@ -142,13 +142,21 @@ func CreateTerragruntEvalContext(
 		"get_terragrunt_source_cli_flag":               wrapVoidToStringAsFuncImpl(getTerragruntSourceCliFlag, extensions.TrackInclude, terragruntOptions),
 	}
 
+	terraformCompatibilityFunctions := map[string]function.Function{
```
Comment (brikis98) [inline config/config_helpers.go:145]: NIT: Why did we need to add these? Add a comment with a link that provides the relevant context.

**Review by brikis98 — DISMISSED** [review]
LGTM!

**Review by zackproser — DISMISSED** [review]
LGTM

**Review by zackproser — APPROVED** [review]
LGTM

**Pull request merged**